tests/foreman/api/test_host.py

```diff
@@ -347,18 +349,26 @@ def test_positive_create_with_puppet_ca_proxy(default_sat):
 
     :CaseImportance: Critical
     """
-    proxy = entities.SmartProxy().search(query={'search': f'url = {default_sat.url}:9090'})[0]
-    host = entities.Host(puppet_ca_proxy=proxy).create()
+    proxy = module_puppet_enabled_sat.api.SmartProxy().search(
+        query={'search': f'url = {module_puppet_enabled_sat.url}:9090'}
+    )[0]
+    host = module_puppet_enabled_sat.api.Host(puppet_ca_proxy=proxy).create()
     assert host.puppet_ca_proxy.read().name == proxy.name
-    new_host = entities.Host().create()
+    new_host = module_puppet_enabled_sat.api.Host().create()
     new_host.puppet_ca_proxy = proxy
     new_host = new_host.update(['puppet_ca_proxy'])
     assert new_host.puppet_ca_proxy.read().name == proxy.name
 
 
 @pytest.mark.tier2
+@pytest.mark.skip_if_open("BZ:2046573")
 def test_positive_end_to_end_with_puppet_class(
-    module_org, module_location, module_env_search, module_puppet_classes, default_smart_proxy
+    module_puppet_org,
+    module_puppet_loc,
+    module_env_search,
+    module_puppet_classes,
+    module_default_puppet_proxy,
```

Comment:
I think we should rename this fixture as to:
```
module_puppet_enabled_proxy
```
or
```
module_puppet_enabled_default_proxy
```

The default word there points to default_sat so puppet enabled will be helpful!

Reply:
it's called `session_puppet_enabled_proxy` now